pwndbg/aglib/saved_register_frames.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import pwndbg.aglib.memory
+
+
+@dataclass
+class SavedRegisterFrame:
+    """
+    A list of registers that have been saved to process memory for later restoration.
+
+    For example, on syscall entry, the process registers are saved to the kernel stack.
+    """
+
+    # List of (offset, register name), sorted from smallest to largest offset
+    frame_layout: list[tuple[int, str]]
+    offsets: dict[str, int]
+
+    def __init__(self, register_offsets: dict[str, int]):
+        self.offsets = register_offsets
+
+        self.frame_layout = sorted([(y, x) for (x, y) in list(register_offsets.items())])
+
+    def read_saved_register(self, reg: str, sp: int = None) -> int | None:
+        if sp is None:
+            sp = pwndbg.aglib.regs.sp
+
+        ptr_size = pwndbg.aglib.arch.ptrsize
+
+        address = sp + self.offsets[reg]
+
+        try:
+            mem = pwndbg.aglib.memory.read(address, ptr_size)
```

Comment:
```suggestion
            mem = pwndbg.aglib.memory.read(address, pwndbg.aglib.arch.ptrsize)
```

The `ptr_size` line above can be removed too

Reply:
We can inline the `address` here too?

Reply:
Done.